From b3630b5369125827bf9ec243149b0f9ed5a5d95a Mon Sep 17 00:00:00 2001
From: Li Wang <liwang@redhat.com>
Date: Mon, 17 Jun 2024 13:34:36 +0800
Subject: [PATCH 1/3] configure.ac: Add _GNU_SOURCE for struct fs_quota_statv
 check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These changes help ensure that the necessary features and definitions
are available during the configuration process, preventing potential
issues related to missing or incompatible definitions. This should
resolve the compilation error related to struct fs_quota_statv:

 /usr/include/xfs/xqm.h:167:8: error: redefinition of ‘struct fs_qfilestatv’

[ pvorel: This is a workaround for xfsprogs issue, can be removed once
change widely adopted
https://lore.kernel.org/linux-xfs/be7f0845-5d5f-4af5-9ca9-3e4370b47d97@sandeen.net/
]

Link: https://lore.kernel.org/ltp/20240617053436.301336-1-liwang@redhat.com/
Reviewed-by: Petr Vorel <pvorel@suse.cz>
Signed-off-by: Li Wang <liwang@redhat.com>
---
 configure.ac | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 85c5739c4..82969b8d3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -181,7 +181,11 @@ AC_CHECK_TYPES([struct file_handle],,,[
 #include <fcntl.h>
 ])
 
-AC_CHECK_TYPES([struct fs_quota_statv],,,[#include <xfs/xqm.h>])
+AC_CHECK_TYPES([struct fs_quota_statv],,,[
+#define _GNU_SOURCE
+#include <xfs/xqm.h>
+])
+
 AC_CHECK_TYPES([struct if_nextdqblk],,,[#include <linux/quota.h>])
 AC_CHECK_TYPES([struct iovec],,,[#include <sys/uio.h>])
 AC_CHECK_TYPES([struct ipc64_perm],,,[#include <sys/ipcbuf.h>])
-- 
2.45.2

